cmd/exporters/prometheus/httpd.go

```diff
@@ -169,35 +173,37 @@ func (p *Prometheus) ServeMetrics(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// filterMetaTags removes duplicate TYPE/HELP tags in the metrics
-// Note: this is a workaround, normally Render() will only add
-// one TYPE/HELP for each metric type, however since some metric
-// types (e.g. metadata_collector_metrics) are submitted from multiple
-// collectors, we end up with duplicates in the final batch delivered
-// over HTTP.
-func filterMetaTags(metrics [][]byte) [][]byte {
-
-	filtered := make([][]byte, 0)
-
-	metricsWithTags := make(map[string]bool)
-
-	for i, m := range metrics {
-		if bytes.HasPrefix(m, []byte("# ")) {
-			if fields := strings.Fields(string(m)); len(fields) > 3 {
-				name := fields[2]
-				if !metricsWithTags[name] {
-					metricsWithTags[name] = true
-					filtered = append(filtered, m)
+// addMetricsToSlice adds metrics to a slice, skipping duplicates. Normally
+// Render() only adds one TYPE/HELP for each metric type. Some metric types
+// (e.g., metadata_collector_metrics) are submitted from multiple collectors.
+// That causes duplicates that are removed in this function. The seen map is
+// used to keep track of which metrics have been added. The metrics slice is
+// expected to be in the format: # HELP metric_name help text # TYPE metric_name
+// type metric_name{tag="value"} value
+func addMetricsToSlice(data [][]byte, metrics [][]byte, seen map[string]bool, addMetaTags bool) [][]byte {
+
+	if !addMetaTags {
+		return append(data, metrics...)
+	}
+
+	for i, metric := range metrics {
```

Comment:
Here, filtering would be on metrics slice, but not on data slice. 
It should be filter data slice as well same as left side ?